### PR TITLE
docs: added documentation on ICS-20 memo field

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

closes: #3008 

### Commit Message / Changelog Entry

```text
docs: added documentation on ICS-20 memo field
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/apps/transfer/messages.md [hunk 1]
```diff
@@ -31,7 +31,26 @@ This message is expected to fail if:
 - `Sender` is empty.
 - `Receiver` is empty.
 - `TimeoutHeight` and `TimeoutTimestamp` are both zero.
+- `Memo` is longer than 256 characters.
```
Comment (colin-axner) [inline docs/apps/transfer/messages.md:34]: I don't think this is true?

File: docs/apps/transfer/messages.md [hunk 2]
```diff
@@ -31,7 +31,26 @@ This message is expected to fail if:
...
 
 This message will send a fungible token to the counterparty chain represented by the counterparty Channel End connected to the Channel End with the identifiers `SourcePort` and `SourceChannel`.
 
 The denomination provided for transfer should correspond to the same denomination represented on this chain. The prefixes will be added as necessary upon by the receiving chain.
+
+### Memo
+
+The memo field was added to allow applications and users to attach metadata to transfer packets. The field is optional and can be left empty. When it is used to attach metadata for a particular middleware, the memo field should be represented as a json object where different middlewares add different json keys.
```
Comment (colin-axner) [inline docs/apps/transfer/messages.md:42]:
```suggestion
The memo field was added to allow applications and users to attach metadata to transfer packets. The field is optional and may be left empty. When it is used to attach metadata for a particular middleware, the memo field should be represented as a json object where different middlewares use different json keys.
```

**Review by colin-axner — APPROVED** [review]

File: docs/apps/transfer/messages.md (hunk 1 above)
Comment (srdtrk) [reply docs/apps/transfer/messages.md:34]: Oh wow, you're right. But it is true for ICA. I'll fix. https://github.com/cosmos/ibc-go/blob/ec3166d31cd86f138e1c5559f29483c348e0b901/modules/apps/27-interchain-accounts/types/packet.go#L49

File: docs/apps/transfer/messages.md (hunk 2 above)
Comment (srdtrk) [reply docs/apps/transfer/messages.md]: done

File: docs/apps/transfer/messages.md (hunk 1 above)
Comment (srdtrk) [reply docs/apps/transfer/messages.md]: Removed this line

**Review by crodriguezvega — APPROVED** [review]
Thank you for adding this, @srdtrk.

**Review by damiannolan — APPROVED** [review]
Thanks @srdtrk

**Review by chatton — APPROVED** [review]
LGTM

**Pull request merged**